src/main/java/mekanism/client/key/MekanismKeyHandler.java

```diff
@@ -63,9 +66,27 @@ public static void registerKeybindings() {
 
     private static void handlePotentialModeItem(EquipmentSlot slot) {
         Player player = Minecraft.getInstance().player;
-        if (player != null && IModeItem.isModeItem(player, slot)) {
-            Mekanism.packetHandler().sendToServer(new PacketModeChange(slot, player.isShiftKeyDown()));
-            SoundHandler.playSound(MekanismSounds.HYDRAULIC);
+        if (player != null) {
+            if (IModeItem.isModeItem(player, slot)) {
+                Mekanism.packetHandler().sendToServer(new PacketModeChange(slot, player.isShiftKeyDown()));
+                SoundHandler.playSound(MekanismSounds.HYDRAULIC);
+            } else if (Mekanism.hooks.CuriosLoaded) {
+                CuriosIntegration.findCurio(player, s -> s.canEquip(slot, player))
+                        .stream()
+                        .filter(r -> {
+                            final boolean isMode = IModeItem.isModeItem(r.stack(), slot);
+                            if (r.stack().getItem() instanceof IGasItem gas) {
+                                return isMode && !gas.getGas(r.stack()).isEmpty();
+                            }
+                            return isMode;
+                        })
```

Comment:
I feel like this may be more readable as two filter calls?
```java
.filter(r -> IModeItem.isModeItem(r.stack(), slot))
.filter(r -> !(r.stack().getItem() instanceof IGasItem item) || !item.getGas(r.stack()).isEmpty())
```

Reply:
That would however create 2 streams, so it'd be less efficient

Reply:
True though it also isn't the most performance critical spot as it is only ran on keybinding press, but even so you could rewrite it to be:
```java
.filter(r -> {
    ItemStack stack = r.stack();
    if (IModeItem.isModeItem(stack, slot)) {
        return !(stack.getItem() instanceof IGasItem item) || !item.getGas(stack).isEmpty();
    }
    return false;
})
```
Which also would have the benefit of removing the performance impact of the instanceof call if it isn't a mode item.